esp/services/ws_workunits/ws_wudetails.cpp

```diff
@@ -517,21 +517,30 @@ void WUDetails::processRequest(IEspWUDetailsRequest &req, IEspWUDetailsResponse
             }
             Owned<ISysInfoLoggerMsgFilter> msgFilter = createSysInfoLoggerMsgFilter();
             msgFilter->setVisibleOnly();
-            Owned<ISysInfoLoggerMsgIterator> msgIter = createSysInfoLoggerMsgIterator(msgFilter);
-            ForEach(*msgIter)
+            try
             {
-                ISysInfoLoggerMsg & sysInfoMsg = msgIter->query();
-                Owned<IEspWUResponseNote> espWuResponseNote = createWUResponseNote("","");
-                StringBuffer tmpbuf;
-                encodeXML(sysInfoMsg.queryMsg(), tmpbuf, ENCODE_NEWLINES, strlen(sysInfoMsg.queryMsg()), true);
-                espWuResponseNote->setSource(sysInfoMsg.querySource());
-                espWuResponseNote->setMessage(tmpbuf.str());
-                espWuResponseNote->setErrorCode(sysInfoMsg.queryLogMsgCode());
-                espWuResponseNote->setSeverity(LogMsgClassToVarString(sysInfoMsg.queryClass()));
-                espWuResponseNote->setCost(0);
-                VStringBuffer idstr("%" I64F "u", sysInfoMsg.queryLogMsgId());
-                espWuResponseNote->setId(idstr.str());
-                espWuResponseNotes.append(*espWuResponseNote.getClear());
+                Owned<ISysInfoLoggerMsgIterator> msgIter = createSysInfoLoggerMsgIterator(msgFilter);
+                ForEach(*msgIter)
+                {
+                    ISysInfoLoggerMsg & sysInfoMsg = msgIter->query();
+                    Owned<IEspWUResponseNote> espWuResponseNote = createWUResponseNote("","");
+                    StringBuffer tmpbuf;
+                    encodeXML(sysInfoMsg.queryMsg(), tmpbuf, ENCODE_NEWLINES, strlen(sysInfoMsg.queryMsg()), true);
+                    espWuResponseNote->setSource(sysInfoMsg.querySource());
+                    espWuResponseNote->setMessage(tmpbuf.str());
+                    espWuResponseNote->setErrorCode(sysInfoMsg.queryLogMsgCode());
+                    espWuResponseNote->setSeverity(LogMsgClassToVarString(sysInfoMsg.queryClass()));
+                    espWuResponseNote->setCost(0);
+                    VStringBuffer idstr("%" I64F "u", sysInfoMsg.queryLogMsgId());
+                    espWuResponseNote->setId(idstr.str());
+                    espWuResponseNotes.append(*espWuResponseNote.getClear());
+                }
+            }
+            catch(IException *e)
+            {
+                // Non-existant global messages may mean that /SysInfoLogs hasn't been created
+                // so catch and ignore the exception
```

Comment:
This exception handler looks too broad, catching and silently ignoring any exception from the block of code above.
Would it make more sense to go around createSysInfoLoggerMsgIterator only?

And/or should createSysInfoLoggerMsgIterator be _not_ throw an error if no messages yet - but instead return an iterator that returns none ?

Reply:
This can be left to a subsequent JIRA/PR. @shamser - please open one and reference/link it to this one.